### PR TITLE
Internet Explorer 11 compatibility

### PR DESCRIPTION
**Pull request by JoseCastilloExperian**

### CODE REVIEW

File: int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js [hunk 1]
```diff
@@ -338,10 +341,14 @@ for (var i = 0; i < buttonSelector.length; i++) {
...
 } else {
-	setEdqInputSelectors();
+	setEdqInputSelectors("");
 }
 if (edqEmailLineSelector) { edqEmailLineSelector.addEventListener("mouseover", function() {enableButtonDisable(edqCurrentSubmitButtonSelector, false);}); }
-if (edqPhoneLineSelectors) { edqPhoneLineSelectors.forEach(phoneSelector => { phoneSelector.addEventListener("mouseover", function() {enableButtonDisable(edqCurrentSubmitButtonSelector, false);}); }); }
+if (edqPhoneLineSelectors) {
+	for (var i = 0; i < edqPhoneLineSelectors.length; i++) {
+		edqPhoneLineSelectors[i].addEventListener("mouseover", function() {enableButtonDisable(edqCurrentSubmitButtonSelector, false);});
+	}
```
Comment (tabeth) [inline int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js]: Don't use C style loops here (especially since the logic you're implementing doesn't require a C style loop). In this particular case, since the issue is specific to a browser, let's use the [polyfill](https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach).

File: int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js [hunk 2]
```diff
@@ -359,16 +366,16 @@ function edqPhoneValidationCallback() {
 	/** TASK:101729 Allow users to continue with invalid phone; 
 	* based on the Business Manager configuration we can set if we want to prevent the user to go through with an invalid phone. */
 	if (edqValidatePhone) {
-		edqPhoneLineSelectors.forEach(phoneSelector => {
-			if (phoneSelector.hasAttribute("edq-metadata")) {
-				var edqPhoneResponse = JSON.parse(phoneSelector.getAttribute("edq-metadata"));
+		for (var i = 0; i < edqPhoneLineSelectors.length; i++) {
```
Comment (tabeth) [inline int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js]: Same as above.

File: int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js [hunk 3]
```diff
@@ -278,16 +278,19 @@ countryDict.push({ key: "AF", value: "AFG" },
 	{ key: "ZW", value: "ZWE" });
 function countryAlpha3(incomingCountryIso2) { 
 	var iso2ToIso3CountryDict;
-	countryDict.forEach((val) => iso2ToIso3CountryDict = (incomingCountryIso2.match(val.key)) ? val.value : iso2ToIso3CountryDict);
-	return iso2ToIso3CountryDict || vDefaultCountry;
+    for (var i = 0; i < countryDict.length; i++) {
```
Comment (tabeth) [inline int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js]: Don't use C style loops here (especially since the logic you're implementing doesn't require a C style loop). In this particular case, since the issue is specific to a browser, let's use the [polyfill](https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach)

File: int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js [hunk 4]
```diff
@@ -492,7 +492,7 @@ function setCheckoutFormEvents() {
 			$("script[src=\"" + edqGlobalIntuitiveUnicornJsPath + "\"]").remove();
 			$("<script>").attr({
 				src:edqGlobalIntuitiveUnicornJsPath,
-				integrity:"sha512-ooVQYWcrVoGAZXC+qPMJaFsEkLB82EsT42J+p0U5INWm+NHrN+XvnWsALGv440Zg3a/QsDJ2L9XR0592fkSBmA==",
+				integrity:"sha512-EclZkBZvchGyhJiTWwD/A2S1gTDzfVterK6Mh/0f176vNrq8KhVZQet7Pf6uRKip+wheIo8K0qndueI5EcPRqQ==",
```
Comment (tabeth) [inline int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js]: This will need to be changed once 1.6.3 is released.

File: int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js [hunk 5]
```diff
@@ -278,7 +278,7 @@ countryDict.push({ key: "AF", value: "AFG" },
 	{ key: "ZW", value: "ZWE" });
 function countryAlpha3(incomingCountryIso2) { 
 	var iso2ToIso3CountryDict;
-	countryDict.forEach((val) => iso2ToIso3CountryDict = (incomingCountryIso2.match(val.key)) ? val.value : iso2ToIso3CountryDict);
+	countryDict.forEach(function(val) { iso2ToIso3CountryDict = (incomingCountryIso2.match(val.key)) ? val.value : iso2ToIso3CountryDict });
```
Comment (tabeth) [inline int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js]: What is the purpose of this change?

File: int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js [hunk 6]
```diff
@@ -287,7 +287,7 @@ function countryAlpha2(incomingCountryIso3) {
 	})[0] || null;
 }
 /*** Set values for EDQ variables ***/
-function setEdqInputSelectors(stageContentLocation = "") {
+function setEdqInputSelectors(stageContentLocation) {
```
Comment (tabeth) [inline int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js]: What’s the purpose of this change?

File: int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js [hunk 7]
```diff
@@ -338,10 +338,10 @@ for (var i = 0; i < buttonSelector.length; i++) {
 if (window.location.href.toLowerCase().match(/checkout/)) {
 	setEdqInputSelectors("shipping");
 } else {
-	setEdqInputSelectors();
+	setEdqInputSelectors("");
 }
```
Comment (tabeth) [inline int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js]: Generally it’s a better practice to set the default on the function level of this is just a default value you’re using here.

File: int_experian/int_experian_sfra/cartridge/templates/default/edq/emailValidation.isml [hunk 8]
```diff
@@ -11,7 +11,7 @@
 <iselseif condition="${dw.system.Site.current.preferences.custom.EDQStaging == 'Production'}">
     <isloop items="${dw.object.SystemObjectMgr.describe('SitePreferences').getCustomAttributeDefinition('EDQProdUrls').values}" var="prod">
         <isif condition="${prod.value == 'EmailValidationJS'}">
-            <script src="${prod.displayValue}" integrity="sha512-WSld0y8PYOWPc/FIbXaD/ojzg354FQ1Rlf/IF2XQYym+ojyk4sKVq/XBIrxBTuhs5hUvs40Yc+VLqrVKjuS9OQ==" crossorigin="anonymous"></script>
+            <script src="${prod.displayValue}" integrity="sha512-cqOaq8x2ZH5Fywf4M5C7DYzBtlPCRZZ/AyFzlQqseyumNeX9MYaVKP3TYWoaxMfDUgeiNf3778mWbbp0TLaeNA==" crossorigin="anonymous"></script>
```
Comment (tabeth) [inline int_experian/int_experian_sfra/cartridge/templates/default/edq/emailValidation.isml]: This is going to need to be changed with 1.6.3.

**Comment by tabeth** [comment]
@JoseCastilloExperian This has been merged right - Let's close this request and merge it in if not.

File: int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js (hunk 4 above)
Comment (JoseCastilloExperian) [reply int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js]: Changed the Integrity keys to match 1.6.3 version!

File: int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js (hunk 6 above)
Comment (tabeth) [reply int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js]: @JoseCastilloExperian - any idea?

File: int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js (hunk 5 above)
Comment (JoseCastilloExperian) [reply int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js]: Compatibility issues with IE11; the syntax "=>" when "foreach" was used was generating troubles to load the file on IE11; using a function syntax is compatible.

File: int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js (hunk 6 above)
Comment (JoseCastilloExperian) [reply int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js]: Compatibility issue with IE11; doesn't allow me to set a default in the function parameters declaration.

File: int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js (hunk 7 above)
Comment (JoseCastilloExperian) [reply int_experian/int_experian_sfra/cartridge/static/default/js/edqUtils.js]: Will change it to use setEdqInputSelectors(); doing a correction in the function

function setEdqInputSelectors(stageContentLocation) {
   stageContentLocation  = stageContentLocation  || "";
   //code
